darshan-util/pydarshan/darshan/experimental/plots/data_access_by_filesystem.py

```diff
@@ -0,0 +1,524 @@
+
+def empty_series_handler(series, filesystem_roots: List[str]):
+    """
+    Parameters
+    ----------
+
+    series: a ``pd.Series`` object
+
+    filesystem_roots: a list of strings containing unique filesystem root paths
+
+    Returns
+    -------
+    A new ``Series`` with any missing filesystem_root indices
+    filled in along with count values of 0 (for plotting purposes).
+    """
+    return series.reindex(filesystem_roots, fill_value=0).astype(np.float64)
+
+
+def rec_to_rw_counter_dfs_with_cols(report: Any,
+                                    file_id_dict: Dict[int, str],
+                                    mod: str = 'POSIX'):
+    """
+    Filter a DarshanReport into two "counters" dataframes,
+    each with read/write activity in each row (at least 1
+    byte read or written per row) and two extra columns
+    for filesystem root path and filename data.
+
+    Parameters
+    ----------
+
+    report: a darshan.DarshanReport()
+
+    file_id_dict: a dictionary mapping integer file hash values
+                  to string values corresponding to their respective
+                  paths
+
+    mod: a string indicating the darshan module to use for parsing
+         (default: ``POSIX``)
+
+
+    Returns
+    -------
+    tuple of form: (df_reads, df_writes)
+    """
+    # filter the DarshanReport into two "counters" dataframes
+    # with read/write activity in each row (at least 1
+    # byte read or written per row)
+    df_reads, df_writes = rec_to_rw_counter_dfs(report=report,
+                                                mod=mod)
+
+    # add columns with filepaths and filesystem root
+    # paths for each event
+    df_reads, df_writes = add_filesystem_cols(df_reads=df_reads,
+                                              df_writes=df_writes,
+                                              file_id_dict=file_id_dict)
+    return df_reads, df_writes
+
+def rec_to_rw_counter_dfs(report: Any,
+                          mod: str = 'POSIX'):
+    """
+    Filter a DarshanReport into two "counters" dataframes,
+    each with read/write activity in each row (at least 1
+    byte read or written per row).
+
+    Parameters
+    ----------
+
+    report: a darshan.DarshanReport()
+
+    mod: a string indicating the darshan module to use for parsing
+         (default: ``POSIX``)
+
+    Returns
+    -------
+    tuple of form: (df_reads, df_writes)
+    """
+    report.mod_read_all_records(mod, dtype='pandas')
+    rec_counters = report.records[mod][0]['counters']
+    
+    # first, filter to produce a dataframe where {mod}_BYTES_READ >= 1
+    # for each row (tracked event for a given rank or group of ranks)
+    df_reads = rec_counters.loc[rec_counters[f'{mod}_BYTES_READ'] >= 1]
+
+    # similar filter for writing
+    df_writes = rec_counters.loc[rec_counters[f'{mod}_BYTES_WRITTEN'] >= 1]
+    return df_reads, df_writes
+
+
+def add_filesystem_cols(df_reads, df_writes, file_id_dict: Dict[int, str]):
```

Comment:
I'm not sure what we have decided for `pandas` type hints, but if nothing else we can add in `pd.DataFrame` for `df_reads` and `df_writes` here.

Reply:
I'd prefer to leave them empty for now--I suggested it was "ok" to leave them in if the developer really wants to, so you didn't have to delete yours, but they are pointless for now really: https://github.com/darshan-hpc/darshan/pull/396#discussion_r670740058